web/src/lib/components/shared-components/settings/setting-accordion-state.svelte

```diff
@@ -35,7 +34,7 @@
         searchParams.delete(queryParam);
       }
 
-      handlePromiseError(goto(`?${searchParams.toString()}`, { replaceState: true, noScroll: true, keepFocus: true }));
+      replaceState(`?${searchParams.toString()}`, { replaceState: true, noScroll: true, keepFocus: true });
```

Comment:
The state passed to `replaceState` doesn't really make sense. I think `goto` is still the right function to use and we should just be more careful about reactivity.

Reply:
Go to here seems to render again the page and cause this effect function to run again, which lead to an infinity loop. Any other suggestion?

Reply:
Removing reactivity from `searchParams` is probably fine:

```ts
const searchParams = new URLSearchParams(page.url.searchParams);

$effect(() => {
  if ($state.size > 0) {
    searchParams.set(queryParam, [...$state].join(' '));
  } else {
    searchParams.delete(queryParam);
  }

  handlePromiseError(goto(`?${searchParams.toString()}`, { replaceState: true, noScroll: true, keepFocus: true }));
});
```

Or an example that includes reactivity for `page.url.searchParams`:

```ts
let search = $derived.by(() => {
  const searchParams = new URLSearchParams(page.url.searchParams);
  if ($state.size > 0) {
    searchParams.set(queryParam, [...$state].join(' '));
  } else {
    searchParams.delete(queryParam);
  }
  return searchParams.toString();
});

$effect(() => {
  handlePromiseError(goto(`?${search}`, { replaceState: true, noScroll: true, keepFocus: true }));
});
```

Reply:
Nice! That indeed fixes it!